File: frontend/lib/models/waste.dart
class Waste {
  final int id;
  final String qrCode;
  final String location;
  final String type;
  final bool recycled;
  final String date;
  final int userId;
  final String userName;
  final int eventId;
  final String eventName;

  Waste({
    required this.id,
    required this.qrCode,
    required this.location,
    required this.type,
    required this.recycled,
    required this.date,
    required this.userId,
    required this.userName,
    required this.eventId,
    required this.eventName,
  });

  factory Waste.fromJson(Map<String, dynamic> json) {
    return Waste(
      id: json['id'] ?? 0,
      qrCode: json['qrCode'] ?? '',
      location: json['location'] ?? '',
      type: json['type'] ?? '',
      recycled: json['recycled'] ?? false,
      date: json['date'] ?? '',
      userId: json['user']?['id'] ?? 0,
      userName: json['user']?['name'] ?? '',
      eventId: json['event']?['id'] ?? 0,
      eventName: json['event']?['name'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'qrCode': qrCode,
      'location': location,
      'type': type,
      'recycled': recycled,
    };
  }
}

File: backend/src/main/java/com/str/backend/model/Waste.java
package com.str.backend.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "waste")
public class Waste {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String qrCode;
    private String location;
    private String type; // "PLÁSTICO", "VIDRIO", "PAPEL", "ORGÁNICO", "OTRO"
    private Boolean recycled;
    private LocalDateTime date;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user; // Usuario que generó el residuo

    @ManyToOne
    @JoinColumn(name = "event_id")
    private Event event;

    public Waste() {}

    // Getters y Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getQrCode() { return qrCode; }
    public void setQrCode(String qrCode) { this.qrCode = qrCode; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public Boolean getRecycled() { return recycled; }
    public void setRecycled(Boolean recycled) { this.recycled = recycled; }

    public LocalDateTime getDate() { return date; }
    public void setDate(LocalDateTime date) { this.date = date; }

    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }

    public Event getEvent() { return event; }
    public void setEvent(Event event) { this.event = event; }
}